x-pack/plugins/data_enhanced/public/search/async_search_strategy.test.ts

```diff
@@ -24,7 +24,11 @@ describe('Async search strategy', () => {
   beforeEach(() => {
     mockCoreSetup = coreMock.createSetup();
     mockDataStart = dataPluginMock.createStartContract();
-    (mockDataStart.search.getSearchStrategy as jest.Mock).mockReturnValue({ search: mockSearch });
+
+    const getStrategy = mockDataStart.search.getSearchStrategy as jest.Mock;
+    getStrategy.mockReset();
```

Comment:
This seems strange that we'd need to reset this mock, but right now it's necessary. What we should really do is change our `searchStartMock` in src/plugins/data/public/search/mocks.ts to a `createSearchStartMock` function, so that you don't leak a mock implementation in one test to another test.